core/src/test/java/kafka/test/ClusterConfig.java

```diff
@@ -219,8 +221,8 @@ public static class Builder {
 
         private Builder() {}
 
-        public Builder setType(Type type) {
-            this.type = type;
+        public Builder setTypes(Set<Type> types) {
+            this.types = Collections.unmodifiableSet(new HashSet<>(types));
```

Comment:
`Collections.unmodifiableSet` will create a new object from the provided argument, so why creating a new `HashSet` and not simply `Collections.unmodifiableSet(types)`?

Reply:
> so why creating a new HashSet and not simply Collections.unmodifiableSet(types)?

We are trying to make `ClusterConfig` immutable. Without copy, users is able to changes the `ClusterConfig`'s `types` by modifying the input `types`. We can simplify the code by `Set.copy` after removing the support of JDK8